### PR TITLE
Add ensureScrollability option to scrollTo command

### PR DESCRIPTION
**Pull request by laistomazz**
- Closes https://github.com/cypress-io/cypress/issues/1924

### User facing changelog

You can now pass the `ensureScrollability: false` option to `.scrollTo()` to skip the scroll ability check of an element. Addresses https://github.com/cypress-io/cypress/issues/1924.

### Additional details

- I've used `ensureScrollability`, but it seems quite a long name for an option. Suggestions are welcome.
- I'll open a PR to update the documentation once or if the PR is approved and we agree on the option naming.

### How has the user experience changed?

Now users can use `scrollTo()` on an element that can have a variable height without getting an error.

#### Before

Screen Shot 2020-07-20 at 10 56 11 AM

#### After

Passing `ensureScrollable: false`

Screen Shot 2020-07-20 at 10 58 14 AM

Passing incorrect arg to `ensureScrollable`

Screen Shot 2020-07-20 at 10 58 34 AM

Trying to scroll unscrollable el without `ensureScrollable: false`

Screen Shot 2020-07-20 at 11 05 51 AM

### PR Tasks

- [x] Have tests been added/updated?
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)?

### CODE REVIEW

**Comment by laistomazz** [comment]
```html
<html>
<body>
</body>
</html>
```

```js
it('check for scrollability', () => {
    cy.visit('index.html')
    cy.scrollTo(0, 500)
})

it('skips scrollability check', () => {
    cy.visit('index.html')
    cy.scrollTo(0, 500, { ensureScrollability: false })
})

it('validates ensureScrollability value', () => {
    cy.visit('index.html')
    cy.scrollTo(0, 500, { ensureScrollability: 'force' })
})
```

ezgif-3-7fba2f3d4c5b

File: packages/driver/src/cy/commands/actions/scroll.js [hunk 1]
```diff
@@ -333,6 +338,10 @@ module.exports = (Commands, Cypress, cy, state) => {
       }
 
       const ensureScrollability = () => {
+        if (!options.ensureScrollability) {
```
Comment (jennifer-shehane) [inline packages/driver/src/cy/commands/actions/scroll.js:341]: Just some comments to clarify + link to original issue.

```suggestion
        // Some elements are not scrollable, user may opt out of error checking
        // https://github.com/cypress-io/cypress/issues/1924
        if (!options.ensureScrollability) {
```

**Comment by laistomazz** [comment]
@jennifer-shehane Thanks for the review and the suggestions.

I initially used `force`, but then I found it a bit weird, because for `cy.click()`, for example, we use `{ force: true }` to click the element even if it's covered by something else (or for other reasons). In this case it will not scroll at all (?), the option would be for the case presented in the issue, for elements with variable content/height.

So I've changed it for `ensureScrollable` for now (until you discuss it with other team members).

I've also added the suggested change for the "not scrollable" error:
Screenshot 2020-07-17 at 12 43 24

File: packages/driver/src/cy/commands/actions/scroll.js [hunk 2]
```diff
@@ -276,6 +277,10 @@ module.exports = (Commands, Cypress, cy, state) => {
         $errUtils.throwErrByPath('scrollTo.invalid_easing', { args: { easing: options.easing } })
       }
 
+      if (!_.isBoolean(options.ensureScrollability)) {
+        $errUtils.throwErrByPath('scrollTo.invalid_ensureScrollability', { args: { ensureScrollability: options.ensureScrollability } })
```
[the remark this answers is not in the document]
Comment (laistomazz) [reply packages/driver/src/cy/commands/actions/scroll.js]: Done :) https://github.com/cypress-io/cypress/pull/8003/commits/7baf159c28c8c4ff114f0009bb1a7d0b873688b9

**Pull request merged**